### PR TITLE
.Net: [Error handling] [Part4] Replacing KernelException and OpenAIInvalidResponseException exceptions by SKException

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
This PR is one of the follow-up PRs aimed at gradually replacing SK custom exceptions with SKException to provide a simple, consistent, and extensible exception model. You can find more details in the [ADR](https://github.com/microsoft/semantic-kernel/blob/main/docs/decisions/0004-error-handling.md) discussing error handling.

### Description
This PR updates a few SK components to throw SKException instead of KernelException and OpenAIInvalidResponseException, and removes those exceptions.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs [hunk 1]
```diff
@@ -69,14 +69,14 @@ private protected ClientBase(ILogger? logger = null)
 
         if (response == null)
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs:70]:
```suggestion
        if (response is null)
```

File: dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs [hunk 2]
```diff
@@ -129,12 +129,12 @@ await foreach (StreamingChoice choice in streamingChatCompletions.GetChoicesStre
 
             if (response == null)
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs:130]:
```suggestion
            if (response is null)
```

File: dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs [hunk 3]
```diff
@@ -109,12 +109,12 @@ public async Task<string> GenerateImageAsync(string description, int width, int
 
         if (result.Result == null)
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs:110]:
```suggestion
        if (result.Result is null)
```

File: dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/TextEmbeddingServiceExtensions.cs [hunk 4]
```diff
@@ -17,12 +18,12 @@ public static class TextEmbeddingServiceExtensions
...
     /// <param name="serviceId">Optional identifier of the desired service.</param>
     /// <returns>The embedding service matching the given id or the default service.</returns>
-    /// <exception cref="KernelException">Thrown when no suitable service is found.</exception>
+    /// <exception cref="SKException">Thrown when no suitable service is found.</exception>
     public static ITextEmbeddingGeneration GetTextEmbeddingService(
         this IAIServiceProvider services,
         string? serviceId = null)
             => services.GetService<ITextEmbeddingGeneration>(serviceId)
-                ?? throw new KernelException(KernelException.ErrorCodes.ServiceNotFound, "Text embedding service not available");
+                ?? throw new SKException("Text embedding service not available");
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/TextEmbeddingServiceExtensions.cs:26]:
```suggestion
                ?? throw new SKException("Text embedding service not found");
```

**Review by RogerBarreto — COMMENTED** [review]
nit. LGFM, small changes suggested.

File: dotnet/src/SemanticKernel.Abstractions/SkillDefinition/NullReadOnlySkillCollection.cs
[the remark this answers is not in the document]
Comment (SergeyMenshykh) [reply dotnet/src/SemanticKernel.Abstractions/SkillDefinition/NullReadOnlySkillCollection.cs:1]: agree, let me see what I can do :)

**Review by RogerBarreto — APPROVED** [review]

**Review by shawncal — APPROVED** [review]

**Pull request merged**